amqp-impl/src/main/java/io/streamnative/pulsar/handlers/amqp/ExchangeContainer.java

```diff
@@ -16,37 +16,37 @@
 
 import com.google.common.collect.Maps;
 import io.streamnative.pulsar.handlers.amqp.impl.PersistentExchange;
-import java.util.HashMap;
 import java.util.Map;
+import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
 import lombok.Getter;
 import lombok.extern.slf4j.Slf4j;
 import org.apache.commons.lang3.StringUtils;
+import org.apache.pulsar.broker.PulsarServerException;
+import org.apache.pulsar.broker.PulsarService;
+import org.apache.pulsar.broker.service.Topic;
 import org.apache.pulsar.broker.service.persistent.PersistentTopic;
 import org.apache.pulsar.common.naming.NamespaceName;
-import org.apache.qpid.server.exchange.ExchangeDefaults;
 
 /**
  * Container for all exchanges in the broker.
  */
 @Slf4j
 public class ExchangeContainer {
 
-    private static final Map<String, AmqpExchange.Type> BUILDIN_EXCHANGE_NAME_MAP = new HashMap<>();
+    private Executor executor = Executors.newCachedThreadPool();
     private AmqpTopicManager amqpTopicManager;
 
-    public ExchangeContainer(AmqpTopicManager amqpTopicManager) {
-        this.amqpTopicManager = amqpTopicManager;
-        BUILDIN_EXCHANGE_NAME_MAP.put(AbstractAmqpExchange.DEFAULT_EXCHANGE_DURABLE, AmqpExchange.Type.Direct);
-        BUILDIN_EXCHANGE_NAME_MAP.put(ExchangeDefaults.DIRECT_EXCHANGE_NAME, AmqpExchange.Type.Direct);
-        BUILDIN_EXCHANGE_NAME_MAP.put(ExchangeDefaults.FANOUT_EXCHANGE_NAME, AmqpExchange.Type.Fanout);
-        BUILDIN_EXCHANGE_NAME_MAP.put(ExchangeDefaults.TOPIC_EXCHANGE_NAME, AmqpExchange.Type.Topic);
+    public ExchangeContainer(AmqpBrokerService amqpBrokerService) {
```

Comment:
Should this constructor be public?  Is it supposed to be created from outside this package?  Or, only through the `AmqpBrokerService`?

Reply:
protected is better :)